helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroCompatibilityHelper.java

```diff
@@ -54,9 +62,16 @@ public class AvroCompatibilityHelper {
           + " differs from constant (" + HelperConsts.HELPER_FQCN + ")");
     }
     DETECTED_VERSION = detectAvroVersion();
+    DETECTED_COMPILER_VERSION = detectAvroCompilerVersion();
     if (DETECTED_VERSION == null) {
       ADAPTER = null;
     } else {
+      if (DETECTED_VERSION != DETECTED_COMPILER_VERSION) {
+        LOGGER.error(
+            String.format("avro and avro-compiler version mismatch! avro version: %s. avro-compiler version: %s",
+                DETECTED_VERSION, DETECTED_COMPILER_VERSION));
+      }
```

Comment:
Not sure if we want to log or throw here...

Reply:
throw, definitely throw.
the most lenient which i still think is reasonable is only throw if/when any compiler-related methods are called, but to begin with i'd just throw.